### PR TITLE
Update llvm_latest to 17 and add llvm_16

### PR DESCRIPTION
**Pull request by nathanchance**
This pull request updates the `llvm_latest` anchor to `17` and adds `llvm_16`.

This adds 1003 builds per week.

-Total builds per week: 8267
+Total builds per week: 9270

Closes: https://github.com/ClangBuiltLinux/continuous-integration2/issues/608

### CODE REVIEW

File: generator.yml [hunk 1]
```diff
@@ -1289,6 +1313,476 @@ builds:
...
+  - {<< : *arm64_fedora,      << : *next,             << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *arm64_suse,        << : *next,             << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *hexagon,           << : *next,             << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *hexagon_allmod,    << : *next,             << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *i386,              << : *next,             << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *i386_suse,         << : *next,             << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *mips,              << : *next,             << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *mipsel,            << : *next,             << : *llvm_full,       boot: true,  << : *llvm_16}
+  # ppc32: Build disabled (https://github.com/ClangBuiltLinux/linux/issues/1679)
+  # - {<< : *ppc32,             << : *next,             << : *llvm,            boot: true,  << : *llvm_16}
```
Comment (nickdesaulniers) [inline generator.yml:1428]: same

File: generator.yml [hunk 2]
```diff
@@ -1289,6 +1313,476 @@ builds:
...
+  - {<< : *arm64_fedora,      << : *mainline,         << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *arm64_suse,        << : *mainline,         << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *hexagon,           << : *mainline,         << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *hexagon_allmod,    << : *mainline,         << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *i386,              << : *mainline,         << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *i386_suse,         << : *mainline,         << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *mips,              << : *mainline,         << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *mipsel,            << : *mainline,         << : *llvm_full,       boot: true,  << : *llvm_16}
+  # ppc32: Build disabled (https://github.com/ClangBuiltLinux/linux/issues/1679)
+  # - {<< : *ppc32,             << : *mainline,         << : *llvm,            boot: true,  << : *llvm_16}
```
Comment (nickdesaulniers) [inline generator.yml:1358]: https://github.com/ClangBuiltLinux/linux/issues/1679 is marked as fixed in llvm-16. Time to re-enable for for this PR?

File: generator.yml [hunk 3]
```diff
@@ -1289,6 +1313,476 @@ builds:
...
+  - {<< : *arm64_fedora,      << : *stable,           << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *arm64_suse,        << : *stable,           << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *hexagon,           << : *stable,           << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *hexagon_allmod,    << : *stable,           << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *i386,              << : *stable,           << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *i386_suse,         << : *stable,           << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *mips,              << : *stable,           << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *mipsel,            << : *stable,           << : *llvm_full,       boot: true,  << : *llvm_16}
+  # ppc32: Build disabled (https://github.com/ClangBuiltLinux/linux/issues/1679)
+  # - {<< : *ppc32,             << : *stable,           << : *llvm,            boot: true,  << : *llvm_16}
```
Comment (nickdesaulniers) [inline generator.yml:1499]: same

File: generator.yml [hunk 4]
```diff
@@ -1289,6 +1313,476 @@ builds:
...
+  - {<< : *arm64_fedora,      << : *stable-6_1,       << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *arm64_suse,        << : *stable-6_1,       << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *hexagon,           << : *stable-6_1,       << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *hexagon_allmod,    << : *stable-6_1,       << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *i386,              << : *stable-6_1,       << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *i386_suse,         << : *stable-6_1,       << : *llvm_full,       boot: false, << : *llvm_16}
+  - {<< : *mips,              << : *stable-6_1,       << : *llvm_full,       boot: true,  << : *llvm_16}
+  - {<< : *mipsel,            << : *stable-6_1,       << : *llvm_full,       boot: true,  << : *llvm_16}
+  # ppc32: Build disabled (https://github.com/ClangBuiltLinux/linux/issues/1679)
+  # - {<< : *ppc32,             << : *stable-6_1,       << : *llvm,            boot: true,  << : *llvm_16}
```
Comment (nickdesaulniers) [inline generator.yml:1569]: same

**Review by nickdesaulniers — COMMENTED** [review]
need to update badges on homepage, too

File: generator.yml (hunk 2 above)
Comment (nathanchance) [reply generator.yml:1358]: Sure, although I will have to do `boot: false` due to https://github.com/ClangBuiltLinux/linux/issues/1814. I'll add a patch on top.

**Comment by nathanchance** [comment]
> need to update badges on homepage, too

https://github.com/ClangBuiltLinux/ClangBuiltLinux.github.io/pull/55

File: generator.yml (hunk 2 above)
Comment (nathanchance) [reply generator.yml]: cfa47bbc73097dfb40f17d68fbf5473652d2acb5

**Review by nickdesaulniers — COMMENTED** [review]
LGTM!

**Comment by nathanchance** [comment]
Huh, got my PRs mixed up... :/ thanks for the post approval heh.

**Pull request merged**